### PR TITLE
More explicit docstring

### PR DESCRIPTION
**Pull request by drewlr**

### CODE REVIEW

**Review by SreeHarshaNelaturu — COMMENTED** [review]
> filter_by_tags: allows to export annotations only for images containing certain image tags. It can be a list of strings or an individual str. 

It can be list of strings or an individual str -> Type: List[str] or str

Thoughts?

**Review by SreeHarshaNelaturu — CHANGES_REQUESTED** [review]
Made some changes. Please take a look.

File: remo/api.py [hunk 1]
```diff
@@ -377,15 +377,19 @@ def export_annotations(
         """
         Exports annotations in a given file format. 
         
-        It offers a choice of convenient export options including appending the full_path to image filenames, choosing between coordinates in pixels or percentages, exporting a separate file for Image tags, and whether to export annotations for a list of images containing certain image tags.
+        It offers multiple convenient format options. In addition to this
```
Comment (drewlr) [inline remo/api.py:380]:
```suggestion
        It offers some convenient export options, including:
```

File: remo/api.py [hunk 2]
```diff
@@ -377,15 +377,19 @@ def export_annotations(
...
+        - Export annotations filtered by user-determined tags.
 
         Args:
             annotation_set_id: annotation set id
             annotation_format: can be one of ['json', 'coco', 'csv']. Default: 'json'
             full_path: if True, appends file path to the filename. uses full image path. Default: True
             export_coordinates: converts output values to percentage or pixels, can be one of ['pixel', 'percent']. Default: 'pixel'
             export_tags: if True, exports the tags to a separate CSV file. Default: True
-            filter_by_tags: allows to export annotations only for images containing certain image tags. It can be a list of strings or an individual str.
+            filter_by_tags: allows to export annotations only for images containing certain image tags. It can be of type List[str] or str. Default: None
```
Comment (drewlr) [inline remo/api.py:392]: Looks good

File: remo/api.py [hunk 3]
```diff
@@ -366,18 +366,31 @@ def get_annotation_set(self, id):
...
+        full_path=True, 
+        export_tags: bool = True, 
+        filter_by_tags: list = None
     ) -> bytes:
         """
-        Exports annotations in given format
+        Exports annotations in a given file format. 
+        
+        It offers some convenient export options, including:
+        - Methods to append the full_path to image filenames,
```
Comment (drewlr) [inline remo/api.py:381]:
```suggestion
        It offers some convenient export options, including:
        
        - Methods to append the full_path to image filenames, 
```
We need a space before a bullet list for it to render correctly on the docs

**Review by drewlr — COMMENTED** [review]
Pending:

- new line before bullet list
- maybe worth adding a one liner example

File: remo/api.py [hunk 4]
```diff
@@ -366,18 +366,33 @@ def get_annotation_set(self, id):
...
+        annotation_format='json', 
+        export_coordinates='pixel', 
+        full_path=True, 
+        export_tags: bool = True, 
+        filter_by_tags: list = None
     ) -> bytes:
+
         """
-        Exports annotations in given format
+        Exports annotations for a given annotation set.
```
Comment (drewlr) [inline remo/api.py:379]:
```suggestion
        Exports annotations for a given annotation set in a given format.
```

File: remo/domain/annotation_set.py [hunk 5]
```diff
@@ -115,17 +115,29 @@ def add_image_annotation(self, image_id: int, annotation: Annotation):
...
+        full_path: bool = True, 
+        export_tags: bool = True, 
+        filter_by_tags: list = None
     ):
         """
-        Exports annotations in a given format
+        Exports annotations for a given annotation set.  
+        
+        It offers some convenient export options, including:
+        - Methods to append the full_path to image filenames,
```
Comment (drewlr) [inline remo/domain/annotation_set.py:129]:
```suggestion
        Exports annotations of the annotation set in a given format.
        
        It offers some convenient export options, including:
        
        - Methods to append the full_path to image filenames, 
```

File: remo/domain/annotation_set.py [hunk 6]
```diff
@@ -148,8 +160,15 @@ def export_annotations_to_file(
         filter_by_tags: list = None
     ):
         """
-        Exports annotations in given format and save to output file
+        
+        Exports annotations in a given format and saves it to a file.
 
+        It offers some convenient export options, including:
+        - Methods to append the full_path to image filenames,
```
Comment (drewlr) [inline remo/domain/annotation_set.py:167]:
```suggestion
        It offers some convenient export options, including:
        
        - Methods to append the full_path to image filenames, 
```

File: remo/domain/dataset.py [hunk 7]
```diff
@@ -248,15 +248,21 @@ def export_annotations(
         filter_by_tags: list = None
     ) -> bytes:
         """
-        Export annotations for a given annotation set
-
+        Export annotations for a given annotation set.
+        
+        It offers some convenient export options, including:
+        - Methods to append the full_path to image filenames,
```
Comment (drewlr) [inline remo/domain/dataset.py:254]:
```suggestion
        Export annotations for a given annotation set in a given format.
        
        It offers some convenient export options, including:
        
        - Methods to append the full_path to image filenames, 
```

File: remo/domain/dataset.py [hunk 8]
```diff
@@ -284,7 +290,13 @@ def export_annotations_to_file(
         filter_by_tags: list = None
     ):
         """
-        Exports annotations in given format and save to output file
+        Exports annotations in a given format and saves it to a file.
+
+        It offers some convenient export options, including:
```
Comment (drewlr) [inline remo/domain/dataset.py:295]:
```suggestion
        Exports annotations for a given annotation set in a given format and saves it to a file.

        It offers some convenient export options, including:
```

File: remo/sdk.py [hunk 9]
```diff
@@ -405,15 +405,21 @@ def export_annotations(
         filter_by_tags: list = None
     ) -> bytes:
         """
-        Exports annotations with the chosen format.
-
+        Exports annotations for a given annotation set.
+        
+        It offers some convenient export options, including:
```
Comment (drewlr) [inline remo/sdk.py:410]:
```suggestion
        Exports annotations for a given annotation set in a given format.
        
        It offers some convenient export options, including:
        
```

File: remo/sdk.py [hunk 10]
```diff
@@ -438,7 +444,25 @@ def export_annotations_to_file(
...
+        Exports annotations in a given format and saves it to a file.
+        
+        It offers some convenient export options, including:
+        - Methods to append the full_path to image filenames, 
+        - Choose between coordinates in pixels or percentages,
+        - Export tags to a separate file
+        - Export annotations filtered by user-determined tags
+
+        Example::
+                # Download and unzip this sample dataset: https://s-3.s3-eu-west-1.amazonaws.com/dogs_dataset.json
```
Comment (drewlr) [inline remo/sdk.py:457]: @SreeHarshaNelaturu this won't render well I think in the docs..?

Comment (SreeHarshaNelaturu) [reply remo/sdk.py:458]: I think it looks fine. Any suggestions on alternative? This is similar to how it is in generate_annotations_from_tags

**Pull request merged**